src/api/hypixel.ts

```diff
@@ -28,20 +37,37 @@ export async function FetchPlayerData(uuid: string) {
 
 	try {
 		const data = await response.json();
-		redis.SETEX(`hypixel:player:${uuid}`, PLAYER_CACHE_TTL, JSON.stringify(data));
+		if (data.success === false) {
+			throw new Error(data.cause || 'Request to Hypixel API failed. Please try again!');
+		}
+
+		if (data.player === null) {
+			throw new Error('Player not found!');
+		}
+
+		const player = data.player;
 
-		return data;
+		redis.SETEX(`hypixel:player:${uuid}`, PLAYER_CACHE_TTL, JSON.stringify(player));
+
+		return player;
 	} catch (error) {
 		console.error(error);
 		return null;
 	}
 }
 
 export async function FetchProfiles(uuid: string) {
+	if (IsUUID(uuid) === false) {
+		const playerUUID = await FetchMinecraftAccountByUsername(uuid);
+
+		uuid = playerUUID?.id || uuid;
+	}
+
 	if (await redis.EXISTS(`hypixel:profiles:${uuid}`)) {
 		const data = await redis.GET(`hypixel:profiles:${uuid}`);
 
-		if (data) {
+		if (data && data !== 'null') {
```

Comment:
No need to check for if data is the string null because we don't store the profiles if we didn't receive any right now